pygmt/tests/test_filter1d.py

```diff
@@ -0,0 +1,96 @@
+"""
+Tests for filter1d.
+"""
+
+import os
+
+import numpy as np
+import pandas as pd
+import pytest
+from pygmt import filter1d
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import GMTTempFile
+from pygmt.src import which
+
+
+@pytest.fixture(scope="module", name="data")
+def fixture_table():
+    """
+    Load the grid data from the sample earth_relief file.
+    """
+    fname = which("@MaunaLoa_CO2.txt", download="c")
```

Comment:
This `@MaunaLoa_CO2.txt` file needs to be added to the cache at https://github.com/GenericMappingTools/pygmt/blob/4b80f3c269f39917c8bb7ee96c5a253279877024/pygmt/helpers/testing.py#L146

Please add it in a separate Pull Request. Also, maybe a good time to implement the seaborn-like sample dataset loader as discussed in #1436.